doc/src/arch/reference.rst

```diff
@@ -1897,6 +1902,98 @@ The clocking configuration is specified with ``<clock>`` tags within the ``<cloc
+    :req_param endy: The end of y grid location of the wire. Value can be relative to the device size.
+    :req_param x: The location of the spine with respect to the x-axis. Value can be relative to the device size.
+    :opt_param repeatx: The horizontal repeat factor of the spine along the device. Value can be relative to the device size.
+    :opt_param repeaty: The vertical repeat factor of the spine along the device. Value can be relative to the device size.
+
+The provided example clock network defines two spines, and neither repeats as each spans the entire height of the device and is locally at the horizontal midpoint of the device.
+
+.. arch:tag:: <rib metal_layer="string" y="integer" startx="integer" endx="integer" repeatx="integer" repeaty="integer"/>
+    :req_param metal_layer: A referenced metal layer that sets the unit resistance and capacitance of the distribution wire over the length of the wire.
+    :req_param startx: The start x grid location, of the wire which runs parallel to the x-axis from startx and ends at endx, inclusive. Value can be relative to the device size.
+    :req_param endx: The end of x grid location of the wire. Value can be relative to the device size.
+    :req_param y: The location of the rib with respect to the y-axis. Value can be relative to the device size.
+    :opt_param repeatx: The horizontal repeat factor of the rib along the device. Value can be relative to the device size.
+    :opt_param repeaty: The vertical repeat factor of the rib along the device. Value can be relative to the device size.
+
+Along each spine and rib is a group of switch points. Switch points are used to describe drive or tap locations along the clock distribution wire:
+
+.. arch:tag:: <switch_point type="{drive | tap}" name="string" yoffset="integer" xoffset="integer" xinc="integer" yinc="integer" buffer="string">
+    :req_param type:
+        * ``drive`` -- Drive points are where the clock distribution wire can be driven by a routing switch or buffer.
+        * ``tap`` --  Tap points are where it can drive a routing switch or buffer to send a signal to a different ``clock_network`` or logicblock.
+    :opt_param xoffset: (Only for ``rib`` network) Offset from the ``startx`` of a rib network.
+    :opt_param yoffset: (Only for ``spine`` network) Offset from the ``starty`` of a spine network.
+    :opt_param xinc: (Only for rib ``tap`` points) Descibes the repeat factor of a series of evenly spaced tap points.
+    :opt_param yinc: (Only for spine ``tap`` points) Descibes the repeat factor of a series of evenly spaced tap points.
+    :req_param buffer: (Required only for ``drive`` points) A reference to a pre-defined routing switch. This switch will be used at the drive point. The clock generator uses two buffers to drive the two portions of this ``clock_network`` wire when it is split at the drive point.
```

Comment:
Probably want a cross-ref to the `<switch>` section so leaders know how these buffers are defined.

The part about two buffers is a bit confusing. I'd suggest adding a diagram to illustrate this. Also in general there may be >2 sub-portions of the clock network, so you may want to rephrase in terms of 'N' buffers where the clock network splits into N parts.

Reply:
The diagram on page 38 of [Mustafa's thesis](https://tspace.library.utoronto.ca/bitstream/1807/97807/3/Abbas_Mustafa_S_201911_MSc_thesis.pdf) shows why there can only be two buffers with the spine/rib model.